package.json

```diff
@@ -35,9 +35,10 @@
   },
   "dependencies": {
     "@ethereumjs/tx": "^3.2.1",
+    "@metamask/eth-sig-util": "^4.0.0",
```

Comment:
Another side-note, Metamask-extension currently has `eth-sig-util@^1.4.2`, `eth-sig-util@^2.0.0`, and `eth-sig-util@^3.0.0` in its `yarn.lock`-file, but it doesn't have a `@metamask/eth-sig-util@^4.0.0`.

Instead of using `require('trezor-connect/lib/plugins/ethereum/typedData')`, we could instead write our own code to supply EIP-712 hashes in `eth-trezor-keyring` using a different library/version.

The plugin is pretty simple, so it would be easy to replicate the code: [trezor-connect/lib/plugins/ethereum/typedData](https://github.com/trezor/connect/blob/develop/src/js/plugins/ethereum/typedData.js)

Reply:
I've done my best to follow and understand this thread.  Based on what i can tell this will fix the issue of being able to sign on a trezor via metamask when transferring out a polygon NFT on opensea?  As right now when you try to transfer a polygon NFT using a trezor T when you click Sign you receive a message "MetaMask Message Signature: Error: Not supported on this device".  How long does the review take to make this in the official build of Metamask? 

anyway we could use this unoffical version to test it out?

Reply:
I've tested this with buying and bidding on NFTs on Polygon on Opensea. I haven't tested it with transferring NFTs on Opensea Polygon, but it should be the same `signTypedData_v4`.

@danfinlay's [mentioned on Twitter](https://twitter.com/danfinlay/status/1490710702067093508) that this might be released on the 14th or 24th of this month.

You're welcome to manually build this yourself ([instructions here](https://github.com/MetaMask/metamask-extension/pull/13358#issuecomment-1021498882)), but be aware that this code hasn't been reviewed by the @Metamask team, so it might contain security issues/bugs, so **only do it if you're aware of the risks**.

Reply:
Hello, i've just want to say that i tried compiling your build @aloisklink following your linked instructions but it doesn't seem to work with signing to Freeze Metadata of an NFT in OpenSea on Polygon chain.
The current build doesn't work literally, while your build it will open the popup to connect/sign with Trezor but it fails with an error.
If you try again it still doesn't work and it doesn't throw anymore the error.

Reply:
I'm getting the same error when using a normal non-Trezor Metamask account too (see https://github.com/MetaMask/metamask-extension/issues/13665#issuecomment-1046391573), so I don't think this error is specific to this PR.

Reply:
> Hello, i just want to say that i tried compiling your build @aloisklink following your linked instructions but it doesn't seem to work with signing to Freeze Metadata of an NFT in OpenSea on Polygon chain.
The current build doesn't work literally, while your build it will open the popup to connect/sign with Trezor but it fails with an error.
If you try again it still doesn't work and it doesn't throw anymore the error.

We are going to follow up with opensea about this, but we believe it to be unrelated to this PR